lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/gateway/ThirdPartyDocumentGateway.java

```diff
@@ -124,6 +153,21 @@ public DocumentCheckResult performDocumentCheck(DrivingPermitForm drivingPermitD
 
         DocumentCheckResult documentCheckResult = responseHandler(httpResponse);
 
+        // Data capture for VC
+        CheckDetails checkDetails = documentCheckResult.getCheckDetails();
```

Comment:
These are all manipulating documentCheckResult but its not clear thats whats happening I would prefer we didnt new CheckDetails and DrivingPermit in the constructor and instead did

        CheckDetails checkDetails = documentCheckResult.getCheckDetails();
        checkDetails.setCheckMethod(OPENID_CHECK_METHOD_IDENTIFIER);
        checkDetails.setIdentityCheckPolicy(IDENTITY_CHECK_POLICY);
        
        DrivingPermit permit = documentCheckResult.getDrivingPermit();
        permit.setIssuedBy(licenceIssuer.toString());
        permit.setDocumentNumber(drivingPermitDocumentNumber);
        permit.setExpiryDate(drivingPermitExpiryDate.toString());

documentCheckResult.setDrivingPermit(permit)
documentCheckResult.setCheckDetails(checkDetails)

Reply:
Removed object creation from constructor to the method